### PR TITLE
Add compatible header for v0.10

### PR DESCRIPTION
**Pull request by eopXD**
This PR is based upon #193. This PR provide users of the legacy (v0.10) intrinsics API compatible headers as a temporary workaround to transit to the newer version of compiler that supports the new (v0.11) intrinsic API.

Only non-overloaded intrinsics are covered here because macro definition does not allow the same token to be resolved differently according to the number of function parameters. An immediate example is shown below.

```cpp
// v0.10
vadd(op1, op2, vl)
vadd(mask, maskedoff, op1, op2, vl)
// Maps to v0.11
__riscv_vadd(op1, op2, vl)
__riscv_vadd_tamu(mask, maskedoff, op1, op2, vl)
```

### CODE REVIEW

**Comment by asb** [comment]
This is great to see!

If this is merged before LLVM 16.0.x is done, please add a link to this header in the Clang release notes that mention the move to v0.11 intrinsics (for this kind of release notes change, you can commit directly to the monorepo's 16.x branch).

**Review by rofirrim — APPROVED** [review]
Given the massive and mechanical changes it is a bit difficult to review it all thoroughly. As a summary of what I understood, this PR 

- adds a new generator for the compatibility header and a decorator for the legacy syntax (along with the required makefile machinery for it)
- adds two new headers `rvv-v0p10-compatible-headers/policy.h` and `rvv-v0p10-compatible-headers/non-policy.h` which redirect to the new prefixed `__riscv_` intrinsics by means of macros (and also swaps the parameters as needed)
- removes the data types section (to be documented elsewhere, I assumed), this causes a number of file renames
- adds extensive testing for the legacy names against the new headers

This all looks sensible to me. Thanks @eopXD for putting this together.

**Review by kito-cheng — APPROVED** [review]

File: rvv-intrinsic-generator/Makefile.api [hunk 1]
```diff
@@ -14,8 +14,8 @@
 # limitations under the License.
 ###############################################################################
 
-CFLAGS?=-O -Werror=implicit-function-declaration
-EXTRA_CFLAGS?=-march=rv64gcv_zvfh -menable-experimental-extensions
+CFLAGS?=-O -Werror=implicit-function-declaration -Wno-incompatible-pointer-types
```
Comment (kito-cheng) [inline rvv-intrinsic-generator/Makefile.api:17]: Wait, why `-Wno-incompatible-pointer-types`?

Comment (eopXD) [reply rvv-intrinsic-generator/Makefile.api]: The test script originally treats some warning with notes as errors. I just modified the script to expose an option to treat warning as error by parsing the last line of the log and remove this addition of `-Wno-incompatible-pointer-types`.

**Comment by eopXD** [comment]
@asb Added into the release note in https://github.com/llvm/llvm-project/commit/d0f4bebc40307399c4a498e055517f9594ac50af.

**Pull request merged**